core/src/main/java/com/linecorp/armeria/server/healthcheck/FixedRateHealthChecker.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the Licenses
+ */
+
+package com.linecorp.armeria.server.healthcheck;
+
+import java.time.Duration;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Supplier;
+
+import com.linecorp.armeria.client.retry.Backoff;
+import com.linecorp.armeria.common.util.AbstractListenable;
+
+import io.netty.channel.EventLoop;
+
+/**
+ * Trigger supplied health checker after constructing and subsequently with the given interval.
+ */
+final class FixedRateHealthChecker extends AbstractListenable<HealthChecker>
```

Comment:
Could we extract the common logic to `ScheduledHealthChecker` or merge `FixedRateHealthChecker` `FixedDelayHealthChecker` into one?

Reply:
Didn't come out with better extraction, so currently decide to merge with flag to adjust next schedule timing. If there is better idea, please :bow:

Reply:
Yeah, the code is small yet. Let's leave it as it is for now.

Reply:
No problem, just curious that how will other member extract, I tried making abstract parent or separate logic, but not fit to my expectation.